Core/include/Acts/Geometry/detail/DetectorVolumeUpdators.hpp

```diff
@@ -0,0 +1,148 @@
+
+  /// @brief a null volume link - explicitely
+  ///
+  /// @note the method parameters are ignored
+  ///
+  inline void update(const GeometryContext& /*ignored*/,
+                     NavigationState& nState) const {
+    nState.currentVolume = dVolume;
+  }
+};
+
+using SingleIndex = std::size_t;
+
+using VariableBoundAxis =
+    Acts::detail::Axis<Acts::detail::AxisType::Variable,
+                       Acts::detail::AxisBoundaryType::Bound>;
+using VariableBoundIndexGrid1 =
+    Acts::detail::Grid<SingleIndex, VariableBoundAxis>;
+
+/// @brief This holds and extracts a collection of detector
+/// volumes. Alternatively an extractor could also use the
+/// detector and its associated detector volume container for
+/// volume extraction.
+///
+struct DetectorVolumesCollection {
+  /// The volumes held by this collection
+  std::vector<const DetectorVolume*> dVolumes = {};
+
+  /// Extract a voume from a collection
+  ///
+  /// @note that geometry context and navigation state are ignored here
+  /// @param index are access indices into the surfaces store
+  ///
+  /// @return the extracted volume
+  inline const DetectorVolume* extract(const GeometryContext& /*gctx*/,
+                                       const NavigationState& /*nState*/,
+                                       const SingleIndex& index) const {
+    return dVolumes[index];
+  }
+};
+
+/// @brief This is used for volumes that are indexed in a bound
+/// 1-dimensional grid, e.g. a z-spaced array, or an r-spaced array
+/// of volumes.
+///
+struct BoundVolumesGrid1Impl : public IManagedDelegateImpl {
+  using IndexedUpdator =
+      IndexedUpdatorImpl<VariableBoundIndexGrid1, DetectorVolumesCollection,
+                         DetectorVolumeFiller>;
+  // The indexed updator
+  IndexedUpdator indexedUpdator;
+
+  /// Allowed constructor with explicit arguments
+  ///
+  /// @param gBoundaries the grid boundaries
+  /// @param bValue the binning value
+  /// @param cVolumes the contained volumes
+  /// @param bTransfrom is the optional transform
+  BoundVolumesGrid1Impl(
+      const std::vector<ActsScalar>& gBoundaries, BinningValue bValue,
+      const std::vector<const DetectorVolume*>& cVolumes,
+      const Transform3& bTransform = Transform3::Identity()) noexcept(false)
+      : indexedUpdator(IndexedUpdator(VariableBoundIndexGrid1(std::make_tuple(
+                                          VariableBoundAxis(gBoundaries))),
+                                      {bValue}, bTransform)) {
+    indexedUpdator.extractor.dVolumes = cVolumes;
+
+    if (gBoundaries.size() != cVolumes.size() + 1u) {
+      throw std::invalid_argument(
+          "DetectorVolumeUpdators: mismatching boundaries and volume numbers");
+    }
+    // Initialize the grid entries
+    for (std::size_t ib = 1u; ib < gBoundaries.size(); ++ib) {
+      indexedUpdator.grid.at(ib) = ib - 1;
+    }
+  }
+  // Deleted default constructor
+  BoundVolumesGrid1Impl() = delete;
```

Comment:
could be dropped as it is implicitly deleted

Reply:
And ones more.